storage/postgres/client.go

```diff
@@ -21,13 +22,54 @@ type Client struct {
 	logger *log.Logger
 }
 
+// pgxLogger is a pgx-compatible logger interface that uses indexer's standard
+// logger as the backend.
+type pgxLogger struct {
+	logger *log.Logger
+}
+
+func (l *pgxLogger) logFuncForLevel(level pgx.LogLevel) func(string, ...interface{}) {
+	switch level {
+	case pgx.LogLevelTrace, pgx.LogLevelDebug:
+		return l.logger.Debug
+	case pgx.LogLevelInfo:
+		return l.logger.Info
+	case pgx.LogLevelWarn:
+		return l.logger.Warn
+	case pgx.LogLevelError, pgx.LogLevelNone:
+		return l.logger.Error
+	default:
+		l.logger.Warn("Unknown log level", "unknown_level", level)
+		return l.logger.Info
+	}
+}
+
+// Implements pgx.Logger interface.
+func (l *pgxLogger) Log(ctx context.Context, level pgx.LogLevel, msg string, data map[string]interface{}) {
```

Comment:
I was going to set up logging just through this: If a query with an error comes through, log it. That doesn't work -- the error message doesn't contain the query. (Which makes sense -- we bubble up those errors anyway)

Then I beefed this up: Every time a query is logged, remember it but don't print it (to reduce logspam). Then when an error is logged, attach the last remembered query to the logline. This didn't work either -- if a query fails, it only produces an error log; the other informational/tracing logs only happen once a query succeeds.

So this part of integration now only takes care of surfacing warnings and errors in our logging system. And the latter sort-of already worked: When pgx generates an error log, it usually (always?) also returns an `error` value from the offending func, and then our code usually (always?) logs based on that error value.

Reply:
> if a query fails, it only produces an error log

Hmm, what is the problem with the remembered query?

Reply:
the error log does not contain the query. I was hoping that for a bad query, we'd see two emitted logs:
1) An info log saying "about to execute query `SELECT foo FROM bar`"; we're memorize the query here
2) An error log saying "bad query; error around `foo`"; we'd display the memorize query at this point

Alas, only step 2 ever happens for bad queries. The info log, when produced, means "query XXX was _successfully_ executed"

Reply:
Gotcha. This makes sense.